indico/modules/categories/controllers/display.py

```diff
@@ -384,3 +386,24 @@ def _checkParams(self):
     def _process(self):
         category_xml_info = XMLCategorySerializer(self.category).serialize_category()
         return Response(category_xml_info, mimetype='text/xml')
+
+
+class RHCategoryCalendarView(RHDisplayCategoryBase):
+    def _process_GET(self):
+        return WPCategory.render_template('display/calendar.html', self.category)
+
+    def _process_POST(self):
+        data = request.json
+        start = dateutil.parser.parse(data['start'])
+        end = dateutil.parser.parse(data['end'])
+        category_events = get_category_timetable([self.category.id], start, end, from_categ=self.category)['events']
```

Comment:
If we only care for events, maybe it'd be a good idea to add a kwarg `only_events` to that function? Getting sessions and contributions is rather expensive and if we don't need them there's no good reason to query them.